src/lint.ts

```diff
@@ -74,19 +76,36 @@ export function lint(
       : path.resolve(options.gtsRootDir, 'tslint.json');
 
     const configuration = Configuration.loadConfigurationFromPath(configPath);
-    const linter = new Linter({ fix, formatter: 'codeFrame' }, program);
 
-    files.forEach(file => {
+    let linter: Linter | undefined = undefined;
+    for (let i = 0; i < files.length; i++) {
+      // only lint MAX_PARALLEL_FILES at one time, this prevents OOM errors
+      // on codebases with large numbers of files:
+      if (i % MAX_PARALLEL_FILES === 0) {
+        if (linter) {
+          const result = linter!.getResult();
+          if (result.errorCount || result.warningCount) {
+            options.logger.log(result.output);
+            return false;
+          }
+        }
+        linter = new Linter({ fix, formatter: 'codeFrame' }, program);
+      }
+      const file = files[i];
```

Comment:
If there are say 27 files - doesn't this mean we'd get to this code block without having instantiated the linter?

Reply:
I don't think so, because `0 % 27` is `0`, so we'll instantiate the linter. Might be worth trying @ofrobots approach of just creating a linter each time, let me try.

Reply:
🤦‍♂ I don't know why my brain read the loop as going down instead of up.  Carry on!

Reply:
It's okay, I think @ofrobots's suggestion of simplifying to creating a linter each time is better any ways.